aptos-move/aptos-vm-types/src/resolver.rs

```diff
@@ -283,3 +283,51 @@ pub trait StateValueMetadataResolver {
         state_key: &StateKey,
     ) -> anyhow::Result<Option<StateValueMetadataKind>>;
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ResourceGroupSize {
+    Concrete(u64),
+    Combined {
+        num_tagged_resources: usize,
+        all_tagged_resources_size: u64,
+    },
+}
+
+pub fn size_u32_as_uleb128(mut value: usize) -> usize {
```

Comment:
Can we have a trivial unit test for this? (probably better than docs, as I couldn't immediately tell what it does)